### ISSUE TITLE
Batch UpdateComputeEnvironment will not update MinvCpus to zero

### ISSUE THREAD

**Issue by cgmartin** [aws/aws-sdk-go-v2#1414]
Confirm by changing [ ] to [x] below to ensure that it's a bug:
- [x] I've gone though the [API reference](https://docs.aws.amazon.com/sdk-for-go/v2/api/)
- [x] I've checked [AWS Forums](https://forums.aws.amazon.com) and [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-go) for answers
- [x] I've searched for [previous similar issues](https://github.com/aws/aws-sdk-go-v2/issues) and didn't find any solution
  
**Describe the bug**
Batch UpdateComputeEnvironment will not update MinvCpus to zero.

**Version of AWS SDK for Go?**
AWS version: 1.8.1

**Version of Go (`go version`)?**
Go version:  go1.16.7

**To Reproduce (observed behavior)**

```go
input := &batch.UpdateComputeEnvironmentInput{
	ComputeEnvironment: computeEnv.ComputeEnvironmentArn,
	ComputeResources: &batchTypes.ComputeResourceUpdate{
		MinvCpus:     0,
		MaxvCpus:     16,
		DesiredvCpus: 4,
	},
	State: batchTypes.CEStateDisabled,
}
output, err := s.BatchClient.UpdateComputeEnvironment(ctx, input)

// Results in the following request body to AWS backend:
// {"computeEnvironment":"arn:aws:batch:us-east-1:...","computeResources":{"desiredvCpus":4,"maxvCpus":16},"state":"DISABLED"}
```
The "minvCpus" property is not sent.

**Expected behavior**

I expect to be able to change the minvCpus value to 0 with an UpdateComputeEnvironment call.

**Additional context**

Serializers are currently implemented to exclude zero values for MinvCpus, MaxvCpus and DesiredvCpus, see https://github.com/aws/aws-sdk-go-v2/blob/a1be7274d533f8612217a3facc79c0b1a902612c/service/batch/serializers.go#L1767-L1780

**Comment by jasdel** [comment]
Thanks for reporting this bug @cgmartin. We're investigating the best way to handle this issue. We think this is similar to the modeling issue the SDK had with Amazon EC2's API model, e.g. https://github.com/aws/aws-sdk-go-v2/issues/1107. 

Similar to the EC2 issue, it looks like the fields should of been modeled as boxed (aka nullable), instead of unboxed, (aka non-nullable). We're investigating if there is a way to fix this issue without making a breaking change to the `batch` module. The `ec2` module had to have a breaking change because there was no other workaround. Its possible this will be needed for `batch` as well.  We're investigating if we can limit the scope of this issue, and address it in the future for other APIs.

**Comment by cgmartin** [comment]
Thank you for reviewing this. FWIW based on the API call, where fields are intended to only be included if they need to change, I agree that it should be modeled as boxed (which was my approach in my local fix #1466). 

And I empathize about the potential breaking change. My expertise with golang is low to confidently suggest a non-breaking path forward. The only idea I have would be to include a completely separate `UpdateComputeEnvironmentInputV2()` function and `batchTypes.ComputeResourceUpdateV2` struct that users could migrate to, though that could potentially cause confusion over the long term.

**Comment by jasdel** [comment]
[service/batch@v1.17.0](https://github.com/aws/aws-sdk-go-v2/blob/main/service/batch/CHANGELOG.md#v1170-2022-03-25) has been updated with a fix that corrects the representation of the members correctly. 

```sh
go get github.com/aws/aws-sdk-go-v2/service/batch@v1.17.0
```

This is a breaking change though. The breaking change was accepted because the API operation could not be used correctly with the members modeled as unboxed (aka value) types. The update changes the members to boxed (aka pointer) types so that the zero value of the members can be handled correctly by the SDK and service.

Your application will fail to compile with the updated module. To workaround this you'll need to update your application to use pointer types for the members impacted.
